Shadows of the Departed/ Apocalypse Nigh

hooksta4 opened the pull request:
<!-- place 'x' mark between square [] brackets to affirm: -->
**_I affirm:_**
- [x] that I agree to Project Topaz's [Limited Contributor License Agreement](http://project-topaz.com/blob/release/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

hooksta4 committed:
Created Shadows of the Departed and Apocalypse_Nigh series of quests. BCNM functions but NM need to be scripted yet.Now with less commits.

wrenffxi commented:
Please don't keep deleting and recreating your fork and PRs.  You only need to fork once, ever, and you only need to create one PR from your branch.  This is your current open PR, so it shouldn't be deleted until it's finished.  Everything that gets messy is fixable without deleting; just have patience and ask for help when needed.  Otherwise you end up leaving a bunch of garbage in the project's commit or PR history.

If you have time this evening, ping me on Discord and I'll help you neaten this PR up.

wrenffxi commented on scripts/globals/bcnm.lua:
```diff
@@ -112,7 +112,7 @@ local battlefields = {
     [36] = {                -- EMPYREAL PARADOX
         { 0, 1056,    0},   -- Dawn (PM8-4)
-     -- { 1, 1057,    0},   -- Apocalypse Nigh (Quest)
+        { 1, 1057,    0},   -- Apocalypse Nigh (Quest)
```
should keep this commented out until the battle is fully coded (i.e. the mobs have their battle mechanics that make the fight challenging)

ibm2431 replied:
Seconding leaving this commented out; Ealdnarche isn't finished.

wrenffxi commented on scripts/zones/Empyreal_Paradox/bcnms/Apocalypse_Nigh.lua:
```diff
@@ -0,0 +1,40 @@
+function onEventFinish(player, csid, option)
+    if csid == 7 then
+        player:setPos(-.0745,-10,-465.1132,63,33);
+        player:setCharVar('ApocalypseNigh',5);
```
Only want to set this charvar on players currently on this quest / at this stage

wrenffxi commented on scripts/zones/Empyreal_Paradox/npcs/Transcendental.lua:
```diff
@@ -18,26 +18,29 @@ end;
     if (player:getCurrentMission(COP) == tpz.mission.id.cop.DAWN and player:getCharVar("PromathiaStatus")==1) then
-        player:startEvent(2);
+        player:startEvent(2)
+    elseif (player:getQuestStatus(JEUNO,tpz.quest.id.jeuno.APOCALYPSE_NIGH) == QUEST_ACCEPTED and player:getCharVar('ApocalypseNigh') == 3) then	
```
Remove unnecessary parentheses.  Because this PR is specifically about adding the questline content, I recommend NOT fixing semicolons and parens on unrelated lines.  But for new lines you're adding, follow the "no unnecessary parentheses" rule, even if the existing file has them.  We'll clean up the existing ones in later, style-specific PRs.

wrenffxi commented on scripts/zones/Empyreal_Paradox/npcs/Transcendental.lua:
```diff
@@ -18,26 +18,29 @@ end;
     if ( csid == 2) then
-        player:setCharVar("PromathiaStatus",2);
+        player:setCharVar("PromathiaStatus",2)
+    elseif (csid == 4) then
```
Unnecessary parens here and elsewhere.  Will stop commenting on them.

wrenffxi commented on scripts/zones/Empyreal_Paradox/npcs/qm1.lua:
```diff
@@ -0,0 +1,45 @@
+-----------------------------------
+require("scripts/globals/missions");
+require("scripts/globals/keyitems");
+
```
unused import of keyitems
missing import of quests, which is used on lines 41 and 42.

wrenffxi commented on scripts/zones/Empyreal_Paradox/npcs/qm1.lua:
```diff
@@ -0,0 +1,45 @@
+require("scripts/globals/missions");
+require("scripts/globals/keyitems");
+
+local Earring =
```
lower camel case variable names.
`local earring =`

also, this variable isn't used anywhere in the script.

wrenffxi commented on scripts/zones/Empyreal_Paradox/npcs/qm1.lua:
```diff
@@ -0,0 +1,45 @@
+
+function onTrigger(player,npc)
+
+    if (player:getCurrentMission(COP) == tpz.mission.id.cop.THE_LAST_VERSE ) and player:hasItem(15962) or player:hasItem(15963) or player:hasItem(15964) or player:hasItem(15965) or player:hasItem(15966)then
```
line-wrap these very long lines, so they're easier for reviewers to see on GitHub.  I usually line break them something like this:

```
    if
        player:getCurrentMission(COP) == tpz.mission.id.cop.THE_LAST_VERSE and 
        (
            player:hasItem(15962) or
            player:hasItem(15963) or
            player:hasItem(15964) or
            player:hasItem(15965) or
            player:hasItem(15966)
        )
    then
        return
    else	
        player:startEvent(5)
    end
```

wrenffxi commented on scripts/zones/Empyreal_Paradox/npcs/qm1.lua:
```diff
@@ -0,0 +1,45 @@
+end;
+
+function onEventUpdate(player,csid,option,extras)
+    if ((csid == 5 and option == 0)) then
```
this doesn't do anything.  can just leave the function empty and get same result

wrenffxi commented on scripts/zones/Empyreal_Paradox/npcs/qm1.lua:
```diff
@@ -0,0 +1,45 @@
+    end
+
+end
+-----------------------------------
```
these function comments aren't needed.  we removed them from most files in the project a while back.

wrenffxi commented on scripts/zones/Grand_Palace_of_HuXzoi/npcs/_iya.lua:
```diff
@@ -26,15 +28,18 @@ function onEventFinish(player,csid,option)
 
     if (csid == 52 and option == 1) then
         player:setPos(-419.995,0,248.483,191,35); -- To The Garden of RuHmet {R}
+	elseif (csid == 4) then 
```
tabs causing weird indentation here

wrenffxi commented on scripts/zones/Lower_Jeuno/npcs/Aldo.lua:
```diff
@@ -40,6 +42,7 @@ function onEventFinish(player,csid,option)
     elseif (csid == 104) then
         player:setCharVar("ZilartStatus",1);
+    elseif (csid == 10057) then
+    player:setCharVar("ApocalypseNigh", 6)
```
missing indentation

wrenffxi commented on scripts/zones/Norg/npcs/Gilgamesh.lua:
```diff
@@ -3,7 +3,10 @@
+local ID = require("scripts/zones/Norg/IDs");
 require("scripts/globals/missions");
+require("scripts/globals/quests");
+require("scripts/globals/keyitems");
```
unused import: keyitems

wrenffxi commented on scripts/zones/Norg/npcs/Gilgamesh.lua:
```diff
@@ -49,13 +55,39 @@ end;
 -- 12 parle de kuzotz ? parle de bijoux aussi
 -- 10 parle de zitah
 function onEventUpdate(player,csid,option)
+    if ((csid == 234 ) and option == 1) then
```
where is event 234 being started?

ibm2431 replied:
Gilgamesh starts event 232 the first time you speak to him; and if you select the option to decide later, he'll start 234 instead. There should be a var set for if the player has already asked to decide later, and Gilgamesh should be choosing the appropriate event based on that.

(Also, option should be 99; option 1 seems to be sent in response to the player choosing an earring)

ibm2431 replied:
You've got event 234 starting now, but will need to have an `or` in the update to account for both 232 and 234

wrenffxi commented on scripts/zones/Norg/npcs/Gilgamesh.lua:
```diff
@@ -49,13 +55,39 @@ end;
+              player:setCharVar("ApocalypseNigh",0)
+          else
+              player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED,reward)
+      end
```
this whole function is improperly indented, and when you fix the indentation it makes some logic errors apparent.  here's what it looks like with fixed indentation:

```
function onEventFinish(player,csid,option)

    if (csid == 99) then
        player:tradeComplete();
        player:setCharVar("MissionStatus",3);
    elseif (csid == 232 or csid == 234) then -- 
        local reward = 0;
        if (option == 1) then
            reward = 15962; -- Static Earring
        elseif (option == 2) then
            reward = 15963; -- Magnetic Earring
        elseif (option == 3) then
            reward = 15964; -- Hollow Earring
        elseif (option == 4) then
            reward = 15965; -- Ethereal Earring
        end

        if (reward ~= 0) then
            if (player:getFreeSlotsCount() >= 1 and player:hasItem(reward) == false) then
                player:addItem(reward);
                player:messageSpecial(ID.text.ITEM_OBTAINED,reward)
                player:completeQuest(JEUNO,tpz.quest.id.jeuno.APOCALYPSE_NIGH)
                player:addMission(COP, tpz.mission.id.cop.THE_LAST_VERSE)
                player:addMission(ZILART,tpz.mission.id.zilart.THE_LAST_VERSE)
            end
            player:setCharVar("ApocalypseNigh",0)
        else
            player:messageSpecial(ID.text.ITEM_CANNOT_BE_OBTAINED,reward)
        end
    end		
end;
```

this would set the player's variable to 0 when they didn't have inventory space for the reward.  it would also give an "Item cannot be obtained" message then reward was 0.

wrenffxi commented on scripts/zones/Promyvion-Dem/npcs/_0id.lua:
```diff
@@ -0,0 +1,29 @@
+-----------------------------------
+local ID = require("scripts/zones/Promyvion-Dem/IDs")
+require("scripts/globals/missions")
+require("scripts/globals/quests")
```
unused import: missions
missing import: keyitems, which is used on line 11 and elsewhere

same for other two fluxes

wrenffxi commented on scripts/zones/Promyvion-Dem/npcs/_0id.lua:
```diff
@@ -0,0 +1,29 @@
+-----------------------------------
+
+function onTrigger(player, npc)
+    if (player:hasKeyItem(tpz.ki.PROMYVION_DEM_SLIVER)) then
```
can join case 1 and 3:

```
function onTrigger(player, npc)
    if
        player:getQuestStatus(JEUNO,tpz.quest.id.jeuno.SHADOWS_OF_THE_DEPARTED) == QUEST_ACCEPTED and
        not player:hasKeyItem(tpz.ki.PROMYVION_DEM_SLIVER)
    then
        player:addKeyItem(tpz.ki.PROMYVION_DEM_SLIVER)
        player:messageSpecial(ID.text.KEYITEM_OBTAINED, tpz.ki.PROMYVION_DEM_SLIVER)
    else
        player:messageSpecial(ID.text.NOTHING_OUT_OF_ORDINARY)
    end
end
```

same for other two fluxes

wrenffxi commented on scripts/zones/Promyvion-Mea/npcs/_0k0.lua:
```diff
@@ -0,0 +1,30 @@
+-----------------------------------
+
+function onTrigger(player, npc)
+if (player:hasKeyItem(tpz.ki.PROMYVION_MEA_SLIVER)) then
```
indentation issues

wrenffxi commented on scripts/zones/RuLude_Gardens/Zone.lua:
```diff
@@ -61,6 +61,12 @@ function onRegionEnter(player, region)
                     player:startEvent(142)
                 elseif player:getQuestStatus(JEUNO, tpz.quest.id.jeuno.STORMS_OF_FATE) == QUEST_ACCEPTED and player:getCharVar('StormsOfFate') == 3 then
                     player:startEvent(143)
+                elseif (player:getQuestStatus(JEUNO,tpz.quest.id.jeuno.SHADOWS_OF_THE_DEPARTED) == QUEST_AVAILABLE and (player:hasKeyItem(tpz.ki.WHISPER_OF_THE_WYRMKING))) then
```
long lines, unnecessary parens

hooksta4 committed:
Update to PR 459

Attempted to fix Wren's fixes and added BCNM related items.
Updates after rebase of trusts

ibm2431 commented on scripts/zones/Empyreal_Paradox/bcnms/Apocalypse_Nigh.lua:
```diff
@@ -0,0 +1,41 @@
+
+function onEventFinish(player, csid, option)
+    if csid == 32001 then 
+        player:setCharVar('ApocalypseNigh',5)
+        if player:getQuestStatus(JEUNO, tpz.quest.id.jeuno.APOCALYPSE_NIGH) == QUEST_ACCEPTED and player:getCharVar('ApocalypseNigh') == 5 then
```
You're setting and then immediately checking for the same char var here~

wrenffxi replied:
Also, this code is setting ApocalypseNigh variable to 5 on all players in the battle, even those not currently on that quest.

ibm2431 commented on scripts/zones/Empyreal_Paradox/bcnms/Apocalypse_Nigh.lua:
```diff
@@ -0,0 +1,41 @@
+require("scripts/globals/battlefield")
+require("scripts/globals/quests")
+-----------------------------------
+-----------------------------------
```
Can remove this second line header

ibm2431 commented on scripts/zones/Empyreal_Paradox/mobs/Ealdnarche.lua:
```diff
@@ -0,0 +1,50 @@
+-----------------------------------
+require("scripts/globals/status")
+require("scripts/globals/magic")
+mixins = {require("scripts/mixins/job_special")}
```
Be sure to remove requires you're not going to use~!

ibm2431 commented on scripts/zones/Empyreal_Paradox/mobs/Ealdnarche.lua:
```diff
@@ -0,0 +1,50 @@
+            m:updateEnmity(target)
+        end
+    end --]]
+end;
```
For new files you're adding, you can go ahead and make sure new semi-colons aren't being introduced

ibm2431 commented on scripts/zones/Empyreal_Paradox/npcs/qm1.lua:
```diff
@@ -0,0 +1,36 @@
+-- NPC:  QM1 
+-----------------------------------
+local ID = require("scripts/zones/Empyreal_Paradox/IDs")
+require("scripts/globals/quests");
```
Semi-colon here

ibm2431 commented on scripts/zones/Empyreal_Paradox/npcs/qm1.lua:
```diff
@@ -0,0 +1,36 @@
+local ID = require("scripts/zones/Empyreal_Paradox/IDs")
+require("scripts/globals/quests");
+
+function onTrade(player,npc,trade)
```
Be sure new commas you're adding have spaces between them:
`function onTrade(player, npc, trade)`

ibm2431 commented on scripts/zones/Empyreal_Paradox/npcs/qm1.lua:
```diff
@@ -0,0 +1,36 @@
+end
+
+function onTrigger(player,npc)
+
```
And to kill these empty lines at the start of functions~

ibm2431 commented on scripts/zones/Empyreal_Paradox/npcs/qm1.lua:
```diff
@@ -0,0 +1,36 @@
+    player:hasItem(15964) or
+    player:hasItem(15965) or 
+    player:hasItem(15966) then
+    player:messageSpecial(ID.text.NOTHING_OUT_OF_ORDINARY)
```
Indent please~!

ibm2431 commented on scripts/zones/Empyreal_Paradox/npcs/qm1.lua:
```diff
@@ -0,0 +1,36 @@
+        player:addMission(COP, tpz.mission.id.cop.DAWN)
+        player:setCharVar('PromathiaStatus',7)
+    end
+end 
```
Also do me a favor: for the new files you're adding, make sure there's a final newline at the end of the line (and that there isn't trailing spaces on the end of lines).

mrhappyasthma recently did a PR to make sure all lines ended with a newline character, so we wanna try to keep them that way! 😄

ibm2431 commented on scripts/zones/Grand_Palace_of_HuXzoi/npcs/_iya.lua:
```diff
@@ -26,15 +28,18 @@ function onEventFinish(player,csid,option)
         player:setPos(-419.995,0,248.483,191,35); -- To The Garden of RuHmet {R}
+        elseif (csid == 4) then 
+            player:setCharVar('ApocalypseNigh',3)
+            player:setPos(-419.995,0,248.483,191,35)
```
Your `elseif` indenting (and subsequent lines) here is one too much~

(Also remember spaces after commas and removing excess parens)

ibm2431 commented on scripts/zones/RuLude_Gardens/Zone.lua:
```diff
@@ -61,6 +61,14 @@ function onRegionEnter(player, region)
                 elseif player:getQuestStatus(JEUNO, tpz.quest.id.jeuno.STORMS_OF_FATE) == QUEST_ACCEPTED and player:getCharVar('StormsOfFate') == 3 then
                     player:startEvent(143)
+                elseif player:getQuestStatus(JEUNO,tpz.quest.id.jeuno.SHADOWS_OF_THE_DEPARTED) == QUEST_AVAILABLE and 
+                    player:hasKeyItem(tpz.ki.WHISPER_OF_THE_WYRMKING) then
```
There should be checks for the two ZM-17 CSes here (Aldo and Gilgamesh). Both are required to flag the quest, and they can be done in either order.

Check for quest completion of Storms of Fate instead of possessing the Whisper KI, if you would~

Wiki also says a game day wait is involved since completing Storms of Fate

(And indenting seems off here)

ibm2431 commented on scripts/zones/RuLude_Gardens/Zone.lua:
```diff
@@ -126,5 +134,19 @@ function onEventFinish(player, csid, option)
         player:setCharVar('StormsOfFate', 0)
+    elseif (csid == 161) then	
+        player:addKeyItem(tpz.ki.NOTE_WRITTEN_BY_ESHANTARL)
+        player:messageSpecial(ID.text.KEYITEM_OBTAINED, tpz.ki.NOTE_WRITTEN_BY_ESHANTARL)
```
You can [use `npcUtil.giveKeyItem` to make these into one call](https://github.com/project-topaz/topaz/blob/release/scripts/globals/npc_util.lua#L284-L293).

ibm2431 commented on scripts/zones/RuLude_Gardens/Zone.lua:
```diff
@@ -126,5 +134,19 @@ function onEventFinish(player, csid, option)
     elseif csid == 143 then
         player:completeQuest(JEUNO, tpz.quest.id.jeuno.STORMS_OF_FATE)
         player:setCharVar('StormsOfFate', 0)
+    elseif (csid == 161) then	
```
These extra parens aren't needed 😉

ibm2431 commented on scripts/zones/RuLude_Gardens/Zone.lua:
```diff
@@ -61,6 +61,14 @@ function onRegionEnter(player, region)
+                elseif player:getQuestStatus(JEUNO,tpz.quest.id.jeuno.SHADOWS_OF_THE_DEPARTED) == QUEST_AVAILABLE and 
+                    player:hasKeyItem(tpz.ki.WHISPER_OF_THE_WYRMKING) then
+                    player:startEvent(161)
+                elseif (player:hasKeyItem(tpz.ki.PROMYVION_HOLLA_SLIVER)) and (player:hasKeyItem(tpz.ki.PROMYVION_MEA_SLIVER)) and (player:hasKeyItem(tpz.ki.PROMYVION_DEM_SLIVER)) then
```
Don't need params around each condition here

ibm2431 commented on scripts/zones/Promyvion-Holla/npcs/_0gc.lua:
```diff
@@ -0,0 +1,28 @@
+        not player:hasKeyItem(tpz.ki.PROMYVION_HOLLA_SLIVER)
+    then
+        player:addKeyItem(tpz.ki.PROMYVION_HOLLA_SLIVER)
+        player:messageSpecial(ID.text.KEYITEM_OBTAINED, tpz.ki.PROMYVION_HOLLA_SLIVER)
```
Again, can use npcUtil to make it only one call

ibm2431 commented on scripts/zones/Promyvion-Holla/npcs/_0gc.lua:
```diff
@@ -0,0 +1,28 @@
+        player:addKeyItem(tpz.ki.PROMYVION_HOLLA_SLIVER)
+        player:messageSpecial(ID.text.KEYITEM_OBTAINED, tpz.ki.PROMYVION_HOLLA_SLIVER)
+    else
+        player:messageSpecial(ID.text.NOTHING_OUT_OF_ORDINARY)
```
Text should be "It appears to be a barrier woven from the energy of overflowing memories..."

ibm2431 commented on scripts/zones/RuLude_Gardens/Zone.lua:
```diff
@@ -61,6 +61,14 @@ function onRegionEnter(player, region)
+                elseif (player:hasKeyItem(tpz.ki.PROMYVION_HOLLA_SLIVER)) and (player:hasKeyItem(tpz.ki.PROMYVION_MEA_SLIVER)) and (player:hasKeyItem(tpz.ki.PROMYVION_DEM_SLIVER)) then
+                    player:startEvent(162)	
+                elseif player:getCharVar("ApocalypseNigh") == 0 and player:getQuestStatus(JEUNO, tpz.quest.id.jeuno.APOCALYPSE_NIGH) == QUEST_AVAILABLE and 
+                    player:getLocalVar('ANZONE') == 0 then
```
Few things:
1) As written, this will trigger for players that haven't done anything. Despite the slightly confusing name of the enum, `QUEST_AVAILABLE` is just zero, which is the default for players who haven't touched a quest - the functions don't check the actual requirements for you (in the current quest system; this will be addressed in the future) 😅 

2) So you'll need to check for completion of Shadows of the Departed

3) There's also a game day wait

4) You don't/shouldn't need to check if the ApocalypseNigh var is zero if you're checking for QUEST_AVAILABLE; the var will _always_ be zero if the quest has never been started or reset, because the var should be cleared upon quest completion

5) And indenting is off 😉

ibm2431 commented on scripts/zones/RuLude_Gardens/Zone.lua:
```diff
@@ -126,5 +134,19 @@ function onEventFinish(player, csid, option)
+        player:delKeyItem(tpz.ki.PROMYVION_HOLLA_SLIVER)
+        player:delKeyItem(tpz.ki.PROMYVION_DEM_SLIVER)
+        player:delKeyItem(tpz.ki.PROMYVION_MEA_SLIVER)
+        player:setCharVar('ApocalypseNigh',0)
```
The `ApocalypseNigh` var starts out at zero, and in the case of resets/repeats, should already be zero (either after the original completion of the quest, or during the reset itself)

ibm2431 commented on scripts/zones/RuLude_Gardens/Zone.lua:
```diff
@@ -126,5 +134,19 @@ function onEventFinish(player, csid, option)
+        player:delKeyItem(tpz.ki.PROMYVION_DEM_SLIVER)
+        player:delKeyItem(tpz.ki.PROMYVION_MEA_SLIVER)
+        player:setCharVar('ApocalypseNigh',0)
+        player:setLocalVar('ANZONE', 1)
```
Be sure to set the gameday wait for Apoc Nigh too~

ibm2431 commented on scripts/zones/Lower_Jeuno/npcs/Aldo.lua:
```diff
@@ -25,6 +25,8 @@ function onTrigger(player,npc)
         player:startEvent(104);
     elseif (ZilartMission == tpz.mission.id.zilart.THE_SEALED_SHRINE and ZilartStatus == 1) then
         player:startEvent(111);
+    elseif player:getQuestStatus(JEUNO,tpz.quest.id.jeuno.APOCALYPSE_NIGH) == QUEST_ACCEPTED and player:getCharVar('ApocalypseNigh') == 5 then
```
bgwiki says there's a rank 5 check as well; which might seem redundant since ZM is supposed to have been completed, but it is possible for players to change nations and reset their rank.

ibm2431 commented on scripts/zones/Lower_Jeuno/npcs/Aldo.lua:
```diff
@@ -40,6 +42,8 @@ function onEventFinish(player,csid,option)
         player:setCharVar("ZilartStatus",1);
+    elseif (csid == 10057) then
+        player:setCharVar("ApocalypseNigh", 6)
+        player:setCharVar("Apoc_Nigh_Reward", os.date("%j")); -- %M for next minute, %j for next day        
```
[`getMidnight` is what you want, which returns the next midnight](https://github.com/project-topaz/topaz/blob/release/scripts/globals/common.lua#L48-L71)

Your later check can then just be `os.time() >= player:getCharVar("Apoc_Nigh_Reward")`

ibm2431 commented on scripts/zones/Norg/npcs/Gilgamesh.lua:
```diff
@@ -17,9 +19,8 @@ function onTrade(player,npc,trade)
 
 function onTrigger(player,npc)
-
+    local realday = tonumber(os.date("%j")); -- %M for next minute, %j for next day
```
When using `getMidnight` with Aldo, this line won't be needed~!

ibm2431 commented on scripts/zones/Norg/npcs/Gilgamesh.lua:
```diff
@@ -38,6 +39,11 @@ function onTrigger(player,npc)
     elseif (ZilartMission == tpz.mission.id.zilart.AWAKENING) then
         player:startEvent(177);
+    elseif (player:getQuestStatus(JEUNO,tpz.quest.id.jeuno.APOCALYPSE_NIGH) == QUEST_ACCEPTED and player:getCharVar('ApocalypseNigh') == 6) 
+        and (realday ~= player:getCharVar("Apoc_Nigh_Reward")) then
```
And as before, `os.time() > player:getCharVar("Apoc_Nigh_Reward")`

ibm2431 commented on scripts/zones/Norg/npcs/Gilgamesh.lua:
```diff
@@ -38,6 +39,11 @@ function onTrigger(player,npc)
         player:startEvent(177);
+    elseif (player:getQuestStatus(JEUNO,tpz.quest.id.jeuno.APOCALYPSE_NIGH) == QUEST_ACCEPTED and player:getCharVar('ApocalypseNigh') == 6) 
+        and (realday ~= player:getCharVar("Apoc_Nigh_Reward")) then
+        player:startEvent(232,234,15962,15963,15964,15965,642854,4095,0)
```
Which of these params are needed? I suspect the last three might not be.

ibm2431 commented on scripts/zones/Norg/npcs/Gilgamesh.lua:
```diff
@@ -49,13 +55,39 @@ end;
-    end
-
-end;
+    elseif (csid == 232) then
```
232 or 234

And if you wrap this in an if check to make sure the option is _not_ the cancel option, you shouldn't need to check if you've assigned a reward down below.

ibm2431 commented on scripts/zones/Norg/npcs/Gilgamesh.lua:
```diff
@@ -49,13 +55,39 @@ end;
+        if (reward ~= 0) then
+            if (player:getFreeSlotsCount() >= 1 and player:hasItem(reward) == false) then
+                player:addItem(reward)
+                player:messageSpecial(ID.text.ITEM_OBTAINED,reward)
```
[npcUtil.giveItem can be used here; it attempts to give an item, and returns true if it succeeded](https://github.com/project-topaz/topaz/blob/release/scripts/globals/npc_util.lua#L185-L195)

```lua
if npcUtil.giveItem(player, reward) then
    player:completeQuest(JEUNO, tpz.quest.id.jeuno.APOCALYPSE_NIGH)
    -- etc
end
```

ibm2431 commented on scripts/zones/Norg/npcs/Gilgamesh.lua:
```diff
@@ -49,13 +55,39 @@ end;
+                player:setCharVar("Apoc_Nigh_Reward",0)
+            end
+        else
+            player:startEvent(233)
```
This doesn't happen; if you choose to cancel the event, Gilgamesh will say the proper lines as part of 232. If he doesn't by default, the client might need an event update in response to the cancel option.

ibm2431 commented on scripts/zones/Norg/npcs/Gilgamesh.lua:
```diff
@@ -38,6 +39,11 @@ function onTrigger(player,npc)
+        and (realday ~= player:getCharVar("Apoc_Nigh_Reward")) then
+        player:startEvent(232,234,15962,15963,15964,15965,642854,4095,0)
+    else
+    	player:startEvent(233)
```
Indenting here is off

ibm2431 commented on sql/mob_groups.sql:
```diff
@@ -1230,8 +1230,8 @@ INSERT INTO `mob_groups` VALUES (32,5536,35,'Ix_aern_drgs_Wynav',0,128,0,0,0,78,
-INSERT INTO `mob_groups` VALUES (3,2184,36,'Kam_lanaut',0,128,0,0,0,75,75,0);
-INSERT INTO `mob_groups` VALUES (4,1158,36,'Eald_narche',0,128,0,0,0,75,75,0);
+INSERT INTO `mob_groups` VALUES (3,6101,36,'Kam_lanaut',0,128,0,10000,1500,75,75,0);
+INSERT INTO `mob_groups` VALUES (4,6100,36,'Eald_narche',0,128,0,5000,2500,75,75,0);
```
Looking at captures, they're both level 78

(And when you have to pick pool IDs, do a favor for the more OCD inclined and have the pool IDs be sequential 😅 )

ibm2431 commented on sql/mob_skill_lists.sql:
```diff
@@ -3461,3 +3461,17 @@ INSERT INTO `mob_skill_lists` VALUES ('Siren',1010,3512); -- Sonic Buffet
 INSERT INTO `mob_skill_lists` VALUES ('Siren',1010,3513); -- Entice
 INSERT INTO `mob_skill_lists` VALUES ('Siren',1010,3514); -- Hysteric Assault
 INSERT INTO `mob_skill_lists` VALUES ('Siren',1010,3515); -- Clarsach Call
+INSERT INTO `mob_skill_lists` VALUES ('Kam_lanaut_2',1011,823);
```
These will cause a merge conflict if they already haven't (my fault, I quick-merged trust ID reservations)

The next free ID after eye-of-the-storm is merged into release (this week) will be 1141

If you're new to Git, I should point out that Git doesn't care about the value of lines, only if they were changed since the last time you tried to edit the same line (in this case, line 3464). It's the "history" that causes the merge conflict; so you'll need to "rebase" on top of current release to resolve the conflict - just adjusting the skill (and later, spell) list IDs won't be enough.

Please let us know if you need help with this!

ibm2431 commented on sql/mob_skill_lists.sql:
```diff
@@ -3461,3 +3461,17 @@ INSERT INTO `mob_skill_lists` VALUES ('Siren',1010,3512); -- Sonic Buffet
+INSERT INTO `mob_skill_lists` VALUES ('Kam_lanaut_2',1011,825);
+INSERT INTO `mob_skill_lists` VALUES ('Kam_lanaut_2',1011,826);
+INSERT INTO `mob_skill_lists` VALUES ('Kam_lanaut_2',1011,827);
+INSERT INTO `mob_skill_lists` VALUES ('Kam_lanaut_2',1011,828);
```
Comment the skill names, please~!

ibm2431 commented on sql/mob_skill_lists.sql:
```diff
@@ -3461,3 +3461,17 @@ INSERT INTO `mob_skill_lists` VALUES ('Siren',1010,3512); -- Sonic Buffet
+INSERT INTO `mob_skill_lists` VALUES ('Kam_lanaut_2',1011,828);
+INSERT INTO `mob_skill_lists` VALUES ('Kam_lanaut_2',1011,829);
+INSERT INTO `mob_skill_lists` VALUES ('Kam_lanaut_2',1011,830);
+INSERT INTO `mob_skill_lists` VALUES ('Eald_Narche_3',1012,986); -- Vortex
+INSERT INTO `mob_skill_lists` VALUES ('Eald_Narche_3',1012,985); -- Stellar Burst
+INSERT INTO `mob_skill_lists` VALUES ('Eald_Narche_3',1012,1006);-- Omega Javelin
+INSERT INTO `mob_skill_lists` VALUES ('Eald_Narche_3',1012,1600);-- Omega Javelin
+INSERT INTO `mob_skill_lists` VALUES ('Eald_Narche_3',1012,988); -- Warp In (Works the same as Ealdnarche's)
+INSERT INTO `mob_skill_lists` VALUES ('Eald_Narche_3',1012,989); -- Warp Out (Works the same as Ealdnarche's)
```
And reorder these so they go up sequentially by skill ID, if you would!

TeoTwawki replied:
since the name field is purely for human readability, how about
 `'EaldNarche (Apoc Nigh)'` ? 
Standard practice is to strip the apostrophes rather than turn them to underscores, and this makes it easier to see which one he is since the 1st two versions of him are in the same zilart mission. and this one is not.

ibm2431 replied:
+1 on stripping and renaming

TeoTwawki replied:
Same for Kamlawhatever :)

ibm2431 commented on sql/mob_skills.sql:
```diff
@@ -1676,7 +1676,7 @@ INSERT INTO `mob_skills` VALUES (1838,1221,'mine_blast',1,10.0,2000,1500,4,24,0,
 -- INSERT INTO `mob_skills` VALUES (1854,1598,'stellar_burst',0,7.0,2000,1500,4,0,0,0,0,0,0);
 -- INSERT INTO `mob_skills` VALUES (1855,1599,'self-destruct',0,7.0,2000,1500,4,0,0,0,0,0,0);
--- INSERT INTO `mob_skills` VALUES (1856,1600,'omega_javelin',0,7.0,2000,1500,4,0,0,0,0,0,0);
+   INSERT INTO `mob_skills` VALUES (1856,1600,'omega_javelin',0,7.0,2000,1500,4,0,0,0,0,0,0);
```
Will this need a skill script?

ibm2431 replied:
Oh, I see it seems to already have a script; I assumed it didn't because it was previously commented out and wasn't being added. 😅

ibm2431 commented on sql/mob_spell_lists.sql:
```diff
@@ -2515,6 +2515,26 @@ INSERT INTO `mob_spell_lists` VALUES ('Ghul-I-Beaban_BLM',305,174,1,255); -- fir
 INSERT INTO `mob_spell_lists` VALUES ('Ghul-I-Beaban_BLM',305,179,1,255); -- blizzaga
 INSERT INTO `mob_spell_lists` VALUES ('Saa_Doyi_the_Fervid',306,320,1,255); -- katon_ichi
 INSERT INTO `mob_spell_lists` VALUES ('Frost_Flambeau',307,180,1,255); -- blizzaga_ii
+INSERT INTO `mob_spell_lists` VALUES ('Eald_Narche_3',308,274,75,255);-- sleepga_ii
```
These spell list IDs will also be a conflict with the recently-merged reservations for trusts; the next free ID is 429

ibm2431 commented on sql/mob_spell_lists.sql:
```diff
@@ -2515,6 +2515,26 @@ INSERT INTO `mob_spell_lists` VALUES ('Ghul-I-Beaban_BLM',305,174,1,255); -- fir
+INSERT INTO `mob_spell_lists` VALUES ('Eald_Narche_3',308,214,75,255);-- flood
+INSERT INTO `mob_spell_lists` VALUES ('Eald_Narche_3',308,176,75,255);-- firaga_iii
+INSERT INTO `mob_spell_lists` VALUES ('Eald_Narche_3',308,181,75,255);-- blizzaga_iii
+INSERT INTO `mob_spell_lists` VALUES ('Eald_Narche_3',308,186,75,255);-- aeroaga_iii
```
The elemental -agas should be listed before the AM (increasing spell IDs), and Sleepga II should be between Flood and Bindga

ibm2431 commented on sql/mob_spell_lists.sql:
```diff
@@ -2515,6 +2515,26 @@ INSERT INTO `mob_spell_lists` VALUES ('Ghul-I-Beaban_BLM',305,174,1,255); -- fir
+INSERT INTO `mob_spell_lists` VALUES ('Kam_Lanaut_2',309,357,75,255);
+INSERT INTO `mob_spell_lists` VALUES ('Kam_Lanaut_2',309,359,75,255);
+INSERT INTO `mob_spell_lists` VALUES ('Kam_Lanaut_2',309,360,75,255);
+INSERT INTO `mob_spell_lists` VALUES ('Kam_Lanaut_2',309,366,76,255);
```
And comment these please~!

ibm2431 reviewed (changes requested):
Thanks for contributing to Project Topaz, hooksta4!

This is your first PR with us, and unless my recollection is failing me, the first time I've reviewed something of yours! Nice to meet you!

I see Wren's already put you through some of the paces (thanks, Wren)! We try to be thorough when reviewing, which can sometimes seem daunting to newer contributors who aren't used to it. 😅 

Please try not to feel overwhelmed or intimidated! Our goal is to help you produce the best code we know how to! If there's questions you have, or help you need, don't be afraid to reach out to us; or to the Discord server - our community strives to help each other!

wrenffxi commented on scripts/globals/bcnm.lua:
```diff
@@ -570,6 +570,7 @@ function checkReqs(player, npc, bfid, registrant)
         [ 993] = function() return ( cop == mi.cop.THE_WARRIOR_S_PATH                                                                                                       ) end, -- PM7-5: The Warrior's Path
         [1024] = function() return ( cop == mi.cop.WHEN_ANGELS_FALL and copStat == 4                                                                                        ) end, -- PM8-3: When Angels Fall
         [1056] = function() return ( cop == mi.cop.DAWN and copStat == 2                                                                                                    ) end, -- PM8-4: Dawn
+        [1057] = function() return ( cop == player:getQuestStatus(JEUNO, tpz.quest.id.jeuno.APOCALYPSE_NIGH) == QUEST_ACCEPTED and player:getCharVar('ApocalypseNigh') == 4 ) end, -- Apocalypse Nigh  
```
`cop == player:getQuestStatus(JEUNO, tpz.quest.id.jeuno.APOCALYPSE_NIGH) == QUEST_ACCEPTED`

should be just

`player:getQuestStatus(JEUNO, tpz.quest.id.jeuno.APOCALYPSE_NIGH) == QUEST_ACCEPTED`

wrenffxi commented on scripts/zones/Empyreal_Paradox/bcnms/Apocalypse_Nigh.lua:
```diff
@@ -0,0 +1,41 @@
+-----------------------------------
```
Naming convention for battlefield luas is: `all_lower_case.lua`

Note: to fix this, don't use Windows to rename the file.  Instead, use your Git client to do it, so Git knows about the rename.  You use TortoiseGit, so you could either:

* right click the file > TortoiseGit > rename

or if you prefer bash command line,

* `git mv <old filename> <new filename>`

hooksta4 committed:
Requested Changes for Shadows of Departed and AN questline

hooksta4 committed:
Fixed some logic problems as well as reenter reqs.

ibm2431 commented on scripts/zones/Empyreal_Paradox/npcs/qm1.lua:
```diff
@@ -0,0 +1,32 @@
+            player:hasItem(15964) or
+            player:hasItem(15965)
+        )
+        then
```
I don't know if we have official style rules for multi-line if checks, but this `then` might be clearer if moved back an indent:

```lua
if
    condition1 and
    condition2
then
    do_this()
else
    do_something_else()
end
```

ibm2431 commented on scripts/zones/Empyreal_Paradox/npcs/qm1.lua:
```diff
@@ -0,0 +1,32 @@
+        player:messageSpecial(ID.text.QM_TEXT)
+    end
+end
+function onEventUpdate(player,csid,option,extras)
```
Blank line between closing `end` and start of next function please~ 😄 

(Through this file)

ibm2431 commented on scripts/zones/Grand_Palace_of_HuXzoi/npcs/_iya.lua:
```diff
@@ -3,40 +3,44 @@
-        player:setPos(-419.995,0,248.483,191,35); -- To The Garden of RuHmet {R}
-    elseif (csid == 1) then
+        player:setPos(-419.995, 0, 248.483, 191,35); -- To The Garden of RuHmet 
+        elseif csid == 4 then
```
Pull the indent on this `elseif` back by one indent

ibm2431 commented on scripts/zones/Norg/npcs/Gilgamesh.lua:
```diff
@@ -38,9 +38,14 @@ function onTrigger(player,npc)
         player:startEvent(177);
+    elseif (player:getQuestStatus(JEUNO,tpz.quest.id.jeuno.APOCALYPSE_NIGH) == QUEST_ACCEPTED and
+        player:getCharVar('ApocalypseNigh') == 6) and
+        player:getCharVar("Apoc_Nigh_Reward") <= os.time() and player:getCharVar('Apoc_Nigh_RewardCS1') == 0 then
```
Break the `then` onto the next line if this is going to be a multi-line `if`

ibm2431 commented on scripts/zones/Empyreal_Paradox/npcs/qm1.lua:
```diff
@@ -0,0 +1,32 @@
+function onEventFinish(player,csid,option)
+    if (csid == 5 and option == 1) then
+        player:delQuest(JEUNO,tpz.quest.id.jeuno.SHADOWS_OF_THE_DEPARTED)
+        player:delQuest(JEUNO,tpz.quest.id.jeuno.APOCALYPSE_NIGH)
```
Remember to delete quest vars too~

(And spaces after commas!)

ibm2431 commented on scripts/zones/RuLude_Gardens/Zone.lua:
```diff
@@ -61,9 +61,19 @@ function onRegionEnter(player, region)
+            elseif player:getQuestStatus(JEUNO,tpz.quest.id.jeuno.SHADOWS_OF_THE_DEPARTED) == QUEST_AVAILABLE 
+            and player:hasCompletedQuest(JEUNO, tpz.quest.id.jeuno.STORMS_OF_FATE)
+            and player:getCharVar("StormsOfFateWait") <= os.time() then
+                player:startEvent (161)
```
Got some spaces between the `player:startEvent` and opening paren to your arguments `(161)` here and elsewhere in section

ibm2431 commented on scripts/zones/RuLude_Gardens/Zone.lua:
```diff
@@ -61,9 +61,19 @@ function onRegionEnter(player, region)
                 end
             end
-        end
+		end
```
Indenting of these ends is off, and might mean there's some logic errors going on

ibm2431 commented on scripts/zones/RuLude_Gardens/Zone.lua:
```diff
@@ -126,5 +136,19 @@ function onEventFinish(player, csid, option)
         player:setCharVar('StormsOfFate', 0)
+		player:setCharVar("StormsOfFateWait",getMidnight())
+    elseif csid == 161 then
+		npcUtil.giveKeyItem (player, tpz.ki.NOTE_WRITTEN_BY_ESHANTARL)
```
There's tabs on these lines instead of spaces~

ibm2431 commented on scripts/zones/RuLude_Gardens/Zone.lua:
```diff
@@ -126,5 +136,19 @@ function onEventFinish(player, csid, option)
+        player:delKeyItem(tpz.ki.PROMYVION_DEM_SLIVER)
+        player:delKeyItem(tpz.ki.PROMYVION_MEA_SLIVER)
+        player:setLocalVar('ANZONE', 1)
+        player:setCharVar("ApocNighWait",getMidnight())	
```
Think you'll want [`getVanaMidnight` (next game day)](https://github.com/project-topaz/topaz/blob/release/scripts/globals/common.lua#L73-L78) here instead of `getMidnight` (server midnight)

ibm2431 commented on scripts/zones/Promyvion-Dem/npcs/_0id.lua:
```diff
@@ -0,0 +1,27 @@
+-----------------------------------
+local ID = require("scripts/zones/Promyvion-Dem/IDs")
+require("scripts/globals/keyitems");
+-----------------------------------
```
Might wanna require the quests global 😉 

(Here and the other fluxes)

ibm2431 commented on scripts/zones/Empyreal_Paradox/bcnms/apocalypse_nigh.lua:
```diff
@@ -0,0 +1,43 @@
+        if player:getQuestStatus(JEUNO, tpz.quest.id.jeuno.APOCALYPSE_NIGH) == QUEST_ACCEPTED and player:getCharVar('ApocalypseNigh') == 5 then
+        player:setCharVar('ApocalypseNigh',5)
+        player:setCharVar("Apoc_Nigh_Reward",getMidnight())
+		player:startEvent(7)
+        else
+            player:startEvent(7)
```
Do both players that are and aren't on the quest get event 7?

Is there an option to skip the event?

ibm2431 commented on scripts/zones/Empyreal_Paradox/bcnms/apocalypse_nigh.lua:
```diff
@@ -0,0 +1,43 @@
+function onEventFinish(player, csid, option)
+    if csid == 32001 then 
+        if player:getQuestStatus(JEUNO, tpz.quest.id.jeuno.APOCALYPSE_NIGH) == QUEST_ACCEPTED and player:getCharVar('ApocalypseNigh') == 5 then
+        player:setCharVar('ApocalypseNigh',5)
+        player:setCharVar("Apoc_Nigh_Reward",getMidnight())
+		player:startEvent(7)
```
Indenting is off

ibm2431 commented on sql/mob_skill_lists.sql:
```diff
@@ -3591,3 +3591,18 @@ INSERT INTO `mob_skill_lists` VALUES ('Siren',1010,3515); -- Clarsach Call
+INSERT INTO `mob_skill_lists` VALUES ('Kamlanaut (Apoc Nigh)',1141,828); -- Water Blade
+INSERT INTO `mob_skill_lists` VALUES ('Kamlanaut (Apoc Nigh)',1141,829); -- Great Wheel
+INSERT INTO `mob_skill_lists` VALUES ('Kamlanaut (Apoc Nigh)',1141,830); -- Light Blade
+INSERT INTO `mob_skill_lists` VALUES ('EaldNarche (Apoc Nigh)',1142,986); -- Vortex
+INSERT INTO `mob_skill_lists` VALUES ('EaldNarche (Apoc Nigh)',1142,985); -- Stellar Burst
```
Hate sounding picky: Stellar Burst (985) should be listed before Vortex (986)

ibm2431 commented on sql/mob_pools.sql:
```diff
@@ -6140,6 +6140,8 @@ INSERT INTO `mob_pools` VALUES (6096,'Frost_Flambeau','Frost_Flambeau',232,0x000
 INSERT INTO `mob_pools` VALUES (6097,'Ophiotaurus','Ophiotaurus',459,0x00006A0B00000000000000000000000000000000,1,1,7,200,100,0,1,1,1,0,0,0,1,129,4,0,0,1,0,1009);
 INSERT INTO `mob_pools` VALUES (6098,'Siren','Siren',44,0x0000320A00000000000000000000000000000000,1,1,7,200,100,0,1,1,1,0,0,32,131,129,0,0,0,0,0,1010);
 INSERT INTO `mob_pools` VALUES (6099,'Nekros_Hound','Nekros_Hound',143,0x00006C0100000000000000000000000000000000,1,1,3,240,100,0,1,0,0,0,0,0,396,133,0,0,0,1,0,143);
+INSERT INTO `mob_pools` VALUES (6100,'Eald_narche_3','Eald_narche_3',150,0x00000B0300000000000000000000000000000000,4,1,9,240,125,0,1,1,0,16,0,32,0,0,0,0,308,1,0,1012);
+INSERT INTO `mob_pools` VALUES (6101,'Kam_lanaut_2','Kam_lanaut_2',351,0x00001D0500000000000000000000000000000000,5,5,3,240,100,0,1,1,1,18,17,32,0,0,0,0,309,1,0,1011);
```
Now that the skill and spell list IDs have been adjusted in `mob_skill_lists` and `mob_spell_list`, they'll need to be updated in the original `mob_pools` too~!

ibm2431 commented on scripts/globals/bcnm.lua:
```diff
@@ -598,6 +599,9 @@ function checkReqs(player, npc, bfid, registrant)
         [ 897] = function() return ( player:hasKeyItem(tpz.ki.WHISPER_OF_THE_WYRMKING)                                                      ) end, -- Quest: The Wyrmking Descends
         [ 928] = function() return ( player:hasCompletedMission(COP, mi.cop.ANCIENT_VOWS) or (cop == mi.cop.ANCIENT_VOWS and copStat >= 2)  ) end, -- Quest: Ouryu Cometh
+        [1057] = function() return ( player:hasCompletedQuest(JEUNO, tpz.quest.id.jeuno.APOCALYPSE_NIGH) or 
+		                             player:getQuestStatus(JEUNO, tpz.quest.id.jeuno.APOCALYPSE_NIGH) == QUEST_ACCEPTED and 
+									 player:getCharVar('ApocalypseNigh') == 4                                                               ) end, -- ApocalypseNigh
```
You actually might need parens here to clarify the order of operations; since
they can enter `if`:
they've completed `or` **(** are on quest `and` on step 4 **)**
```lua
    player:hasCompletedQuest(JEUNO, tpz.quest.id.jeuno.APOCALYPSE_NIGH) or 
    (player:getQuestStatus(JEUNO, tpz.quest.id.jeuno.APOCALYPSE_NIGH) == QUEST_ACCEPTED and 
    player:getCharVar('ApocalypseNigh') == 4)
```

wrenffxi replied:
It's okay as coded.  AND happens before OR.  That said, parens don't hurt in cases like this, where it makes the order of operations clear to people reading it in the future.
*edit: Oh, this is outdated code.  Leaving comment for posterity.*

ibm2431 reviewed (changes requested):
I think I've gone through and marked all previous comments which have now been resolved as being resolved.

If a thread isn't resolved, then I think it's still an issue~

hooksta4 committed:
Updates to Shadows of Departed and Apoc Nigh

Updates based on comments

hooksta4 committed:
Update bcnm.lua

My dumbass was rushing through and forgot to delete 3 copied lines.

wrenffxi committed:
Wren's fixes for Hooksta's ApocNigh PR

hooksta4 committed:
Merge pull request #1 from wrenffxi/ApocNigh

Wren's fixes for Hooksta's ApocNigh PR

wrenffxi commented:
Notes from the latest commit

### General changes:

* Some style stuff (tabs -> spaces, fix indendation, add newline at bottom of files)
* Some missing includes
* Some post-quest-stage NPC dialog

### Shadows of the Departed:

* Quest now requires player to have finished Awakening
* Fixed default message on Memory Fluxes
* Fixed Memory Fluxes in Promyvion-Mea to match retail captures
* Added post-event message to "trade in three slivers" event

### Apocalypse Nigh:

* Rename Transcendental Radiance script
* Reordered Kam'lanaut and Eald'narch pools and lists
* Reassigned skill and spell lists
* Removed ancient magic from Eald'narch's spell list
* Set levels and immunities.
* Eald'narch now teleports
* Kam'lanaut now uses en-element weapon skills
* Apoc Nigh now progresses and warps characters who are on the correct stage of quest
* Midnight wait happens on Gilgamesh's door, rather than on Gilgamesh.
* Fixed options and parameters on "get earring" events
* Memory Reset NPC now sets progress so that it matches retail capture

### What's left to do:

* Kam'lanaut and Eald'narch are facing the wrong way when players enter arena.
* Kam'lanaut "Can perform three TP attacks upon reaching 100%+ TP, giving very little rest for healers. He can do 3 Great Wheels in a row, devastating any melee jobs in range."
* Kam'lanaut has trait "When using enspell TP moves, nukes of the matching element heal him."
* Eald'narch is "Highly evasive. Sushi and/or Madrigal recommended."
* Eald'narch has "Very high magic defense (e.g., Thunder IV ~65 dmg)."

Once these points are addressed, I think we can toggle the battlefield on in bcnm.lua.

Otherwise, I believe the rest of the questline is working/retail accurate and could merge into Topaz.

ibm2431 committed:
Minor fixes related to Apocalypse Nigh

ibm2431 commented:
Thanks for the assist, Wren! 👍 

(I'm embarrassed I spaced checking and handling Zilart progress)

Pushed a small commit with my final nitpicks:
- Commented Ealdnarche 2's moves since the file diff from your rename brought them up. Saw he had the skills "ark_guardian_tarutaru" (936) and "tarutaru_warp_ii" (962). Went out on a limb and assumed it was someone trying to handle warping by copying TT and removed the skills now that there's Warp In and Warp Out.
- On that note, Ealdnarche's skill list was the only one using those two skills, so they are now orphaned.
- Cleared the var after the midnight wait between Shadows of the Departed and Apocalypse Nigh
- Overly pick spell list order correction

ibm2431 reviewed (approved):
Nice work, team!

Pool IDs will probably change post-merge into canary/release due to Limbus reservations, but that's my problem to worry about. 😃
